benchexec/tools/tbf.py

```diff
@@ -0,0 +1,101 @@
+"""
+BenchExec is a framework for reliable benchmarking.
+This file is part of BenchExec.
+
+Copyright (C) 2007-2018  Dirk Beyer
+All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+import benchexec.result as result
+import benchexec.util as util
+import benchexec.tools.template
+import logging
+import os
+
+from benchexec.model import SOFTTIMELIMIT
+
+
+class Tool(benchexec.tools.template.BaseTool):
+    """
+    Tool info for tbf (https://github.com/sosy-lab/tbf).
+    """
+
+    REQUIRED_PATHS = ["tbf", "lib"]
+
+    def program_files(self, executable):
+        files = super().program_files(executable)
+
+        if 'bin/' in executable:
+            # Get parent of directory of `executable`
+            base_dir = os.path.dirname(os.path.dirname(executable))
+
+        return files + [os.path.join(base_dir, p) for p in self.REQUIRED_PATHS]
+
+    def executable(self):
+        return util.find_executable('tbf', 'bin/tbf')
+
+    def version(self, executable):
+        stdout = self._version_from_tool(executable)
+        return stdout.strip()
```

Comment:
`_version_from_tool` also calls `strip`, so this can be simplified to `return self._version_from_tool(executable)`.

Reply:
Thanks!